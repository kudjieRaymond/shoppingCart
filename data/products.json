[
  {
    "id": 1,
    "sku": 18644119330491312,
    "title": "Sphynx Tie Dye Grey T-Shirt",
    "brand": "Sphynx",
    "description": "Sphynx Tie Dye Grey",
    "availableSizes": ["X", "L", "XL", "XXL"],
    "price": 10,
    "isFreeShipping": true,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  },

  {
    "id": 2,
    "sku": 11854078013954528,
    "brand": "DGK",
    "title": "Danger Knife Grey T-Shirt",
    "description": "Danger Knife Grey",
    "availableSizes": ["X", "M", "L"],
    "price": 14.9,
    "isFreeShipping": true,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  },

  {
    "id": 3,
    "sku": 876661122392077,
    "title": "White DGK Script Tee",
    "description": "White DGK Script",
    "availableSizes": ["X", "M", "L"],
    "price": 14.9,
    "isFreeShipping": true,
    "quantity": 10,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  },

  {
    "id": 4,
    "sku": 9197907543445677,
    "title": "Born On The Streets  T-Shirt",
    "description": "Born On The Streets",
    "availableSizes": ["XL"],
    "price": 25.9,
    "isFreeShipping": false,
    "quantity": 10,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  },

  {
    "id": 5,
    "sku": 10547961582846888,
    "title": "Tso 3D Short Sleeve T-Shirt A",
    "description": "Tso 3D Short Sleeve",
    "availableSizes": ["X", "L", "XL"],
    "price": 10.9,
    "isFreeShipping": false,
    "quantity": 15,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  },

  {
    "id": 6,
    "sku": 6090484789343891,
    "title": "Man Tie Dye Cinza Grey T-Shirt",
    "description": "Man Tie Dye Cinza Grey",
    "availableSizes": ["XL", "XXL"],
    "price": 49.9,
    "isFreeShipping": false,
    "quantity": 5,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  },

  {
    "id": 7,
    "sku": 18532669286405342,
    "title": "Crazy Monkey Black T-Shirt",
    "description": "1977 Infantil",
    "availableSizes": ["S"],
    "style": "Preto com listras brancas",
    "price": 22.5,
    "isFreeShipping": true,
    "quantity": 7,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  },

  {
    "id": 8,
    "sku": 5619496040738316,
    "title": "Tso 3D Black T-Shirt",
    "description": "",
    "availableSizes": ["XL"],
    "style": "Azul escuro",
    "price": 18.7,
    "isFreeShipping": false,
    "quantity": 8,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  },
  {
    "id": 9,
    "sku": 11600983276356165,
    "title": "Crazy Monkey Grey",
    "description": "",
    "availableSizes": ["L", "XL"],
    "price": 134.9,
    "isFreeShipping": true,
    "quantity": 14,
    "thumbnail_url": "https://www.dropbox.com/s/78fot6w894stu3n/img3.jpg?raw=1"
  }
]
