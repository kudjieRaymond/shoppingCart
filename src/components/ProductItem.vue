<template>
  <v-flex xs12 sm6 md4 lg4>
    <v-card>
      <div>
        <v-img :src="item.thumbnail_url"></v-img>
      </div>

      <v-card-text>
        <div class="title my-2">
          <v-btn text small color="green" :to="{name:'product', params:{id:item.id}}">{{item.title}}</v-btn>
        </div>
        <div class="subheading">{{ item.quantity }} left in stock</div>
        <div class="text--primary">{{ item.description | shortDescription}}</div>
      </v-card-text>

      <v-card-actions class="d-flex justify-end">
        <v-btn color="primary" :disabled="item.quantity=== 0"  @click="addItem">Add to cart</v-btn>
      </v-card-actions>
    </v-card>
  </v-flex>
</template>
<script>
import  {mapActions} from 'vuex'
export default {
  props: ['item'],

  //data() {},
  methods: {
		...mapActions(['updateCart'] ),
    addItem(){
			let order = {
				item: Object.assign({}, this.item),
				quantity: 1,
				isAdd:true
			}
			this.updateCart(order)
		}
  },
  filters: {
    shortDescription(value) {
      if (value && value.length > 100) {
        return value.substring(0, 100) + '...';
      }
      return value;
    }
  }
};
</script>