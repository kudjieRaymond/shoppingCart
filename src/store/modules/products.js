import { ref } from '@/firebase';

const products = {
  //namespaced: true,

  state: {
    isLoading: true,
    productList: [
      // {
      //   id: 1,
      //   title: 'macbook Retina 13.3" ME662 (2013)',
      //   thumbnail_url:
      //     'https://www.dropbox.com/s/6tqcep7rk29l59e/img2.jpeg?raw=1',
      //   price: 999.9,
      //   quantity: 10,
      //   category: 'macbook',
      //   description:
      //     '3.0GHz Dual-core Haswell Intel Core i5 Turbo Boost up to 3.2 GHz, 3MB L3 cache 8GB (two 4GB SO-DIMMs) of 1600MHz DDR3 SDRAM'
      // },
      // {
      //   id: 2,
      //   title: 'Macbook Pro 13.3" Retina MF841LL/A',
      //   thumbnail_url:
      //     'http://media.bizwebmedia.net//sites/72783/data/images/2015/11/3220113retina13.jpg',
      //   price: 1199.9,
      //   quantity: 15,
      //   category: 'macbook',
      //   description:
      //     'Macbook Pro 13.3" Retina MF841LL/A Model 2015 Option Ram Care 12/2016'
      // },
      // {
      //   id: 3,
      //   title: 'Macbook Pro 15.4" Retina MC975LL/A Model 2012',
      //   thumbnail_url:
      //     'https://www.dropbox.com/s/6tqcep7rk29l59e/img2.jpeg?raw=1',
      //   price: 1800,
      //   quantity: 1,
      //   category: 'macbook',
      //   description:
      //     '3.0GHz Dual-core Haswell Intel Core i5 Turbo Boost up to 3.2 GHz, 3MB L3 cache 8GB (two 4GB SO-DIMMs) of 1600MHz DDR3 SDRAM'
      // },
      // {
      //   id: 4,
      //   title: 'Retina MacBook Pro 13 inch MF841',
      //   thumbnail_url:
      //     'http://media.bizwebmedia.net//sites/72783/data/images/2016/2/4713895macbook_pro_retina.png',
      //   price: 1000,
      //   quantity: 0,
      //   category: 'macbook',
      //   description:
      //     '2.9 Ghz Dual-Core Intel Core i5 Broadwell Tubro boost up to 3.3 GHz with L3 3MB cache'
      // }
    ]
  },
  getters: {
    products(state) {
      return state.productList;
    },
    isProductLoading(state) {
      return state.isLoading;
    }
  },
  mutations: {
    updateProductList(state, productList) {
      state.productList = productList;
      state.isLoading = false;
    }
  },
  actions: {
    updateProductList({commit}) {
			ref.child("products").on('value', function(snapshot) {
				commit('updateProductList', snapshot.val());
      });
      
    }
  }
};

export default products;
