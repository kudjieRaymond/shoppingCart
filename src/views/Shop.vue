<template>
  <v-container grid-list-lg>
    <div class="d-flex justify-center" v-if="isLoading">
      <ring-loader :loading="isProductLoading" :height="loaderHeight" :color="loaderColor" :size="loaderSize"></ring-loader>
    </div>
    <v-layout row wrap>
      <product-item v-for="prod in products" :item="prod" :key="prod.id"></product-item>
    </v-layout>
  </v-container>
</template>
<script>
import ProductItem from '@/components/ProductItem';
import { mapGetters } from 'vuex';
import RingLoader from 'vue-spinner/src/RingLoader.vue';

export default {
  components: {
    ProductItem,
    RingLoader
  },
  data() {
    return {
      loaderColor: '#5cb85c',
			loaderSize: '100px',
			loaderHeight: '100px',
      isLoading: true
    };
  },

  computed: {
    ...mapGetters(['products', 'isProductLoading'])
  }
};
</script>