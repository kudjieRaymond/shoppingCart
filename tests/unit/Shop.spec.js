import Vuetify from 'vuetify';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Shop from '@/views/Shop'
import Vuex from 'vuex'
import products from '@/store/modules/products';
import RingLoader from 'vue-spinner/src/RingLoader.vue';
import ProductItem from '@/components/ProductItem';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Store.vue', ()=>{

	let state;
  let store;
	let vuetify;

  beforeEach(() => {
		vuetify = new Vuetify();
    state = {
				isLoading: true,
				productList :	 [
					{
					id: 1,
					title: 'macbook Retina 13.3" ME662 (2013)',
					thumbnail_url:
						'https://www.dropbox.com/s/6tqcep7rk29l59e/img2.jpeg?raw=1',
					price: 999.9,
					quantity: 10,
					description:
						'3.0GHz Dual-core Haswell Intel Core i5 Turbo Boost up to 3.2 GHz, 3MB L3 cache 8GB (two 4GB SO-DIMMs) of 1600MHz DDR3 SDRAM'
				},
				{
					id: 2,
					title: 'Macbook Pro 13.3" Retina MF841LL/A',
					thumbnail_url:
						'http://media.bizwebmedia.net//sites/72783/data/images/2015/11/3220113retina13.jpg',
					price: 1199.9,
					quantity: 15,
					description:
						'Macbook Pro 13.3" Retina MF841LL/A Model 2015 Option Ram Care 12/2016'
				}
			]
		}

		store = new Vuex.Store({
      modules: {
        products: {
          state,
          getters: products.getters
        }
      }
    });
	})

	const mountFunction = options => {
    return shallowMount(Shop, {
			store,
      localVue,
      vuetify,
      ...options
    });
  };
		
		it('display a list of products', ()=>{
			const wrapper = mountFunction()
			expect(wrapper.html()).toMatchSnapshot();
			
		})

		it('display a loader', ()=>{
			const wrapper = mountFunction();
			expect(wrapper.find(RingLoader).exists()).toBe(true);
		})

		it('display products', () => {
      const wrapper = mountFunction();
      expect(wrapper.find(ProductItem).exists()).toBe(true);
    });

});
	
	