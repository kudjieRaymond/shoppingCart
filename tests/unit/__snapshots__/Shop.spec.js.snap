// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store.vue display a list of products 1`] = `
<v-container grid-list-lg="">
  <div class="d-flex justify-center">
    <ring-loader-stub loading="true" color="#5cb85c" size="100px" margin="2px" radius="100%" height="100px"></ring-loader-stub>
  </div>
  <v-layout row="" wrap="">
    <product-item-stub item="[object Object]"></product-item-stub>
    <product-item-stub item="[object Object]"></product-item-stub>
  </v-layout>
</v-container>
`;
